open_ce/graph.py

```diff
@@ -0,0 +1,42 @@
+import networkx as nx
+
+class OpenCEGraph(nx.DiGraph):
+    def __init__(self, incoming_graph_data=None, **attr):
+        super().__init__(incoming_graph_data, **attr)
+
+    def add_node(self, node_for_adding, **attr):
+        if node_for_adding in self.nodes():
```

Comment:
Might be faster to do something like: `existing_node = next(x for x in self.nodes() if x == node_for_adding, None)`

Reply:
Yeah, I like doing it that way better. It looks cleaner too.